core/src/main/java/nl/nn/adapterframework/compression/MessageZipEntry.java

```diff
@@ -0,0 +1,89 @@
+/*
+   Copyright 2023 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.compression;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import org.apache.logging.log4j.Logger;
+
+import nl.nn.adapterframework.stream.Message;
+import nl.nn.adapterframework.stream.MessageContext;
+import nl.nn.adapterframework.util.LogUtil;
+import nl.nn.adapterframework.util.MessageUtils;
+import nl.nn.adapterframework.util.StreamUtil;
+
+public class MessageZipEntry extends ZipEntry {
+	private Logger log = LogUtil.getLogger(MessageZipEntry.class);
+	private final Message message;
+
+	public MessageZipEntry(Message message) {
+		this(message, String.valueOf(message.getContext().get(MessageContext.METADATA_NAME)));
+	}
+
+	public MessageZipEntry(Message message, String filename) {
+		super(filename);
+
+		if (message!=null) {
+			this.message = message;
+
+			setSize(message.size());
```

Comment:
Why `message.size()` instead of `computeSize()`?

Reply:
Size is an optional field, if it's unknown and not required, there is no need to compute it.